5.4.connected_vessels.py

```diff
@@ -0,0 +1,23 @@
+def interleave(*iterables):
+    """
+    Interleaves the items of multiple iterables into a single list.
+    :param iterables: unknown number of iterable objects.
+    :return: A list of items interleaved from the input iterables
+    """
+    result_list = []
+    [result_list.extend(zip_item) for zip_item in list(zip(*iterables))]
```

Comment:
Don't use list comprehention for the side effect - 
https://stackoverflow.com/questions/5753597/is-it-pythonic-to-use-list-comprehensions-for-just-side-effects